core/src/main/java/hudson/PluginManager.java

```diff
@@ -819,6 +819,7 @@ public void dynamicLoad(File arc) throws IOException, InterruptedException, Rest
      */
     @Restricted(NoExternalUse.class)
     public void dynamicLoad(File arc, boolean removeExisting) throws IOException, InterruptedException, RestartRequiredException {
+        try (ACLContext context = ACL.as(ACL.SYSTEM)) {
```

Comment:
Maybe it is reasonable to keep the entire method within the clause anyway. There are extension points like `PluginStrategy#updateDependency()`, which may also blow up in the case of the non-System user